controllers/storage/storagepolicy/storagepolicy_controller.go

```diff
@@ -72,19 +72,39 @@ type Reconciler struct {
 	vmProvider providers.VirtualMachineProviderInterface
 }
 
-// +kubebuilder:rbac:groups=storage.k8s.io,resources=storageclasses,verbs=get;list;watch
+// +kubebuilder:rbac:groups=vsphere.policy.vmware.com,resources=storagepolicies,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=vsphere.policy.vmware.com,resources=storagepolicies/status,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Closest place I can comment: Line 66 should replace `StorageClass` with `StoragePolicy` object?